lib/whitehall_importer/create_file_attachment_revision.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module WhitehallImporter
+  class CreateFileAttachmentRevision
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(whitehall_file_attachment)
+      @whitehall_file_attachment = whitehall_file_attachment
+    end
+
+    def call
+      download_file
+    end
+
+  private
+
+    attr_reader :whitehall_file_attachment
+
+    def download_file
+      file = URI.parse(whitehall_file_attachment["url"]).open
+      if file.is_a?(StringIO)
+        # files less than 10 KB return StringIO (we have to manually cast to a tempfile)
+        Tempfile.new.tap { |tmp| File.write(tmp.path, file.string) }
```

Comment:
Worth adding something to the commit message that this method is copied from another class?
And a note that we should look at extracting it out to its own helper later?

Reply:
Agree with you, I implemented it's own class and then saw Kev's comment so reverted to how it was before just without the StringIO bit as it's actually not needed for file attachments.